src/hooks/useOutsideClick .tsx

```diff
@@ -0,0 +1,19 @@
+import { useEffect, RefObject } from 'react';
+
+const useOutsideClick = (ref: RefObject<HTMLElement>, callback: () => void) => {
+  const handleClick = (e: any) => {
+    if (ref.current && !ref.current.contains(e.target)) {
+      callback();
+    }
+  };
+
+  useEffect(() => {
+    document.addEventListener('click', handleClick);
+
+    return () => {
+      document.removeEventListener('click', handleClick);
+    };
+  });
```

Comment:
shouldnt probably add listener to every render? add []

Reply:
I guess shouldn't add listener It works precisely